airflow/contrib/operators/s3_to_sftp_operator.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from airflow.models import BaseOperator
+from airflow.hooks.S3_hook import S3Hook
+from airflow.contrib.hooks.ssh_hook import SSHHook
+from tempfile import NamedTemporaryFile
+from urllib.parse import urlparse
+
+
+def get_s3_key(s3_key):
+    """This parses the correct format for S3 keys
+        regardless of how the S3 url is passed."""
+
+    parsed_s3_key = urlparse(s3_key)
+    return parsed_s3_key.path.lstrip('/')
+
+
+class S3ToSFTPOperator(BaseOperator):
+    """
+    S3 To SFTP Operator
+    :param sftp_conn_id:    The sftp connection id.
+    :type sftp_conn_id:     string
+    :param sftp_path:       The sftp remote path.
+    :type sftp_path:        string
+    :param s3_conn_id:      The s3 connnection id.
+    :type s3_conn_id:       string
+    :param s3_bucket:       The targeted s3 bucket.
+    :type s3_bucket:        string
+    :param s3_key:          The targeted s3 key.
+    :type s3_key:           string
+    """
+
+    template_fields = ('s3_key', 'sftp_path')
+
+    def __init__(self,
+                 sftp_conn_id=None,
```

Comment:
Wouldn't it be better to have default values other than None? Think of this.. Would it work to use it like this with no values being set or shouldn't be there a default behaviour?

I personally would like to have a default s3_conn_id that will be used when I do not pass one like `s3_default` or sth like this.

Reply:
Yes, either a useful default or no default at all (making it a required argument).